packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -350,6 +361,100 @@ export class SeenGossipBlockInput {
     //   blockInputMeta: {pending: GossipedInputType.block, haveBlobs: blobsCache.size, expectedBlobs: null},
     // };
   }
+
+  private hasSampledDataColumns(dataColumnCache: DataColumnsCacheMap): boolean {
+    return (
+      dataColumnCache.size >= this.custodyConfig.sampledColumns.length &&
+      this.custodyConfig.sampledColumns.reduce((acc, columnIndex) => acc && dataColumnCache.has(columnIndex), true)
+    );
+  }
+
+  private async reconstructColumns(config: ChainForkConfig, blockRoot: RootHex) {
```

Comment:
What do you think about breaking the meat of this out into a free function with roughly this signature

```ts
async function reconstructColumns(executionEngine: IExecutionEngine, blockInput: BlockInput): Promise<fulu.DataColumnSidecars>;
```

That was we can reuse the logic in the `unknownBlock` flow to make the call a second time after the gossip wait/delay period (when `getBlobsV1` is called now).

Reply:
As a note, I'm also open to merging this as-is and then we can do that refactor in a later PR where we implement the second call.  Up to you on how you feel is best to proceed.

Reply:
Let's do this in a later PR. I'll start working on it now, but I don't want it to hold up this PR if it's okay!